### PR TITLE
feat(rest): print curl

### PR DESCRIPTION
**Pull request by kobenguyent**
## Motivation/Description of the PR
- That's very high chance you need to share the request for debugging purpose, and curl would be the cheapest and fastest way for that purpose. Hence, printing the curl request in console logs would benefit us.

```console
    › [CURL Request] curl --location --request POST https://httpbin.org/post -H "X-API-Key: apiKey-onRequest" -H "Content-Type: application/json" -d '{"_overheadLength":161,"_valueLength":0,"_valuesToMeasure":[{"fd":null,"path":"/Users/t/Desktop/projects/codeceptjs-rest-demo/src/fixtures/test_image.png","flags":"r","mode":438,"end":null,"bytesRead":0,"_readableState":{"state":6176,"highWaterMark":65536,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":false,"errored":null,"defaultEncoding":"utf8","awaitDrainWriters":null,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3}],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":false,"_streams":["----------------------------807986065015258526651649\r\nContent-Disposition: form-data; name=\"attachment\"; filename=\"test_image.png\"\r\nContent-Type: image/png\r\n\r\n",{"source":{"fd":null,"path":"/Users/t/Desktop/projects/codeceptjs-rest-demo/src/fixtures/test_image.png","flags":"r","mode":438,"end":null,"bytesRead":0,"_readableState":{"state":6176,"highWaterMark":65536,"buffer":{"head":null,"tail":null,"length":0},"length":0,"pipes":[],"flowing":false,"errored":null,"defaultEncoding":"utf8","awaitDrainWriters":null,"decoder":null,"encoding":null},"_events":{},"_eventsCount":3},"dataSize":0,"maxDataSize":null,"pauseStream":true,"_maxDataSizeExceeded":false,"_released":false,"_bufferedEvents":[{"0":"pause"}],"_events":{},"_eventsCount":1},null],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------807986065015258526651649"}'
```

Applicable helpers:
- [ ] REST

## Type of change
- [ ] :rocket: New functionality

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by kobenguyent** [comment]
ready for testing: codeceptjs@3.6.4-beta.2

File: lib/helper/REST.js [hunk 1]
```diff
@@ -372,3 +375,22 @@ class REST extends Helper {
...
+      curl += `-H "${key}: ${value}" `;
+    });
+  }
+
+  if (!curl.toLowerCase().includes('content-type: application/json')) {
+    curl += '-H "Content-Type: application/json" ';
+  }
+
+  if (request.data) {
+    curl += `-d '${JSON.stringify(request.data)}'`;
```
Comment (DavertMik) [inline lib/helper/REST.js:396]: What if data is not JSON? But encoded form?

**Review by DavertMik — CHANGES_REQUESTED** [review]
Looks really good to me
I checked alternative implementation at https://github.com/anthonygauthier/axios-curlirize/blob/master/src/lib/CurlHelper.js and I think your is better yet minimal.

However, I can assume it doesn't cover all cases.

Should we inform users somehow on that?

Or I have a better idea, ignore CURL output if request is not JSON. I don't think we need to fit for all cases

File: lib/helper/REST.js (hunk 1 above)
Comment (kobenguyent) [reply lib/helper/REST.js:396]: that won't be a problem with encoded form, the curl will look like this.

```bash
    › [CURL Request] curl --location --request POST https://httpbin.org/post -H "X-API-Key: apiKey-onRequest" -H "Content-Type: application/json" -d '{"_overheadLength":161,"_valueLength":0,"_valuesToMeasure":[{"fd":null,"path":"/Users/t/Desktop/projects/codeceptjs-rest-demo/src/fixtures/test_image.png","flags":"r","mode":438,"end":null,"bytesRead":0,"_events":{},"_readableState":{"highWaterMark":65536,"buffer":[],"bufferIndex":0,"length":0,"pipes":[],"awaitDrainWriters":null},"_eventsCount":3}],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":false,"_streams":["----------------------------917245260724904416987426\r\nContent-Disposition: form-data; name=\"attachment\"; filename=\"test_image.png\"\r\nContent-Type: image/png\r\n\r\n",{"source":{"fd":null,"path":"/Users/t/Desktop/projects/codeceptjs-rest-demo/src/fixtures/test_image.png","flags":"r","mode":438,"end":null,"bytesRead":0,"_events":{},"_readableState":{"highWaterMark":65536,"buffer":[],"bufferIndex":0,"length":0,"pipes":[],"awaitDrainWriters":null},"_eventsCount":3},"dataSize":0,"maxDataSize":null,"pauseStream":true,"_maxDataSizeExceeded":false,"_released":false,"_bufferedEvents":[{"0":"pause"}],"_events":{},"_eventsCount":1},null],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------917245260724904416987426"}'
```

**Comment by kobenguyent** [comment]
> Looks really good to me I checked alternative implementation at https://github.com/anthonygauthier/axios-curlirize/blob/master/src/lib/CurlHelper.js and I think your is better yet minimal.
> 
> However, I can assume it doesn't cover all cases.
> 
> Should we inform users somehow on that?
> 
> Or I have a better idea, ignore CURL output if request is not JSON. I don't think we need to fit for all cases

updated as proposed!

**Pull request merged**